exercises/practice/binary-search-tree/.meta/lib/example.dart

```diff
@@ -4,39 +4,35 @@ class BinarySearchTree<T extends Comparable<T>> {
 
   BinarySearchTree(T rootData) : root = new Node<T>(rootData) {
     // Note: [assert] does not run in release mode
```

Comment:
Are you sure about removing this? I believe you added it for a reason.

Reply:
The intention was to make sure `rootData` is never null. Since you can't perform comparision on null.

This can be achieved by changing the type constraint from
```
class BST<T> {}
```
To
```
class BST<T extends Object> {}
```

And because the constructor initialises the `root` we don't need a body for the constructor.

Reply:
> If this doesn't work we'll have to create a separate issue to track this.

Yeah, there's a lot I may be missing. For now, it should be fine. In the tests we are declaring that left and right won't be `null`, so that's good for now to direct students around that.